ts/Dashboards/EditMode/AccordionMenu.ts

```diff
@@ -288,6 +336,114 @@ class AccordionMenu {
         }
         return;
     }
+
+    /**
+     * Closes the sidebar discarding changes. If there are any changes, it will
+     * show a confirmation popup. If no changes, it will close the sidebar.
+     */
+    public cancelChanges(): void {
+        if (Object.keys(this.changedOptions).length < 1) {
+            this.closeSidebar();
+        } else {
+            this.showCancelConfirmationPopup();
+        }
+    }
+
+    /**
+     * Confirms changes made in the component.
+     *
+     * @fires EditMode#componentChanged
+     */
+    private async confirmChanges(): Promise<void> {
+        const component = this.component;
+        if (!component) {
+            return;
+        }
+
+        if (Object.keys(this.chartOptionsJSON).length) {
+            await component.update({
```

Comment:
This is only true for the Highcharts component right? Wouldn't it be possible to improve the typing?